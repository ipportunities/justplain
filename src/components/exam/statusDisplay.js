import React from 'react'
import { useTranslation } from '../translate'

const statusColorMap = {
  'generated': 'primary',
  'checked': 'success',
  'filled': 'warning',
  'overdue': 'danger',
  'started': 'info'
}

export const ExamMomentStatusDisplay = ({ status }) => {
  const t = useTranslation()
  if (status) {
    return <h5>
      <span className={ `badge badge-pill badge-${ statusColorMap[status] }` }>
        { t(status) }
      </span>
    </h5>
  } else {
    return <div />
  }
}