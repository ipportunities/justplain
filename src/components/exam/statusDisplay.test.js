import React from 'react'
import { shallow } from 'enzyme'
import { ExamMomentStatusDisplay } from './statusDisplay'

//   'generated': 'primary',
//   'checked': 'success',
//   'filled': 'warning',
//   'overdue': 'danger',
//   'started': 'info'

describe('Exam Moment Status Display', () => {
  it('should render correctly with status generated', () => {
    const component = shallow(<ExamMomentStatusDisplay status='generated' />)
    expect(component).toMatchSnapshot()
  })
  it('should render correctly with status checked', () => {
    const component = shallow(<ExamMomentStatusDisplay status='checked' />)
    expect(component).toMatchSnapshot()
  })
  it('should render correctly with status filled', () => {
    const component = shallow(<ExamMomentStatusDisplay status='filled' />)
    expect(component).toMatchSnapshot()
  })
  it('should render correctly with status overdue', () => {
    const component = shallow(<ExamMomentStatusDisplay status='overdue' />)
    expect(component).toMatchSnapshot()
  })
  it('should render correctly with status started', () => {
    const component = shallow(<ExamMomentStatusDisplay status='started' />)
    expect(component).toMatchSnapshot()
  })

  it('should return div when no status is given', () => {
    const component = shallow(<ExamMomentStatusDisplay />)
    expect(component).toMatchSnapshot()
  })
})
