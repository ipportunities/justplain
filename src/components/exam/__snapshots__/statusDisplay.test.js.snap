// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam Moment Status Display should render correctly with status checked 1`] = `
<h5>
  <span
    className="badge badge-pill badge-success"
  >
    checked
  </span>
</h5>
`;

exports[`Exam Moment Status Display should render correctly with status filled 1`] = `
<h5>
  <span
    className="badge badge-pill badge-warning"
  >
    filled
  </span>
</h5>
`;

exports[`Exam Moment Status Display should render correctly with status generated 1`] = `
<h5>
  <span
    className="badge badge-pill badge-primary"
  >
    generated
  </span>
</h5>
`;

exports[`Exam Moment Status Display should render correctly with status overdue 1`] = `
<h5>
  <span
    className="badge badge-pill badge-danger"
  >
    overdue
  </span>
</h5>
`;

exports[`Exam Moment Status Display should render correctly with status started 1`] = `
<h5>
  <span
    className="badge badge-pill badge-info"
  >
    started
  </span>
</h5>
`;

exports[`Exam Moment Status Display should return div when no status is given 1`] = `<div />`;
